### DISCUSSION TITLE
np.concatenate is taking long time and using very high memory

### DISCUSSION THREAD

**Discussion by galib36** [brianhie/scanorama#95] [Q&A]
Hello,
I was using `scanorama` to integrate spatial transcriptomic data with single cell RNA-seq data. I used the following command for the integration
```
adata_all_integrated = adata_snRNA.concatenate(
    adata_spatial,
    batch_key="dataset",
    batch_categories=["droplet", "visium"],
    join="outer",
    uns_merge="first",
)
```
But when I run,
```
embedding_all = np.concatenate(integrated, axis=0)
```
it takes very long time and uses all the memory in the machine (64GB memory). I had to stop it in the middle. Can you please let me know how to solve this issue.

Thank you.

**Comment by alessiavalenti** [comment]
Hi, 
I'm facing the same problem. Did you solve it or have more insights about it?

**Comment by galib36** [comment]
No, I could not solve it yet.

**Comment by brianhie** [comment]
Sorry for the delay in responding -- the integration command you posted doesn't seem to call scanorama? If you are worried about high memory, try the scanorama.integrate() function, which only operates in the low dimensional space and usually performs better than the high-dimensional correction functionality.
